core/src/main/java/com/linecorp/armeria/server/healthcheck/DiskMemoryHealthChecker.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.io.File;
+
+/**
+ * A {@link HealthChecker} that reports as unhealthy
+ * when the target free disk space exceeds a file disk usable space.
+ * For example:
+ * <pre>{@code
+ * final DiskMemoryHealthChecker diskMemoryHealthChecker = HealthChecker.ofDisk(0.8, new File("/tmp"));
+ *
+ * // Returns false if a file disk usable memory space is less than 80%,
+ * // or true if a file disk usable memory space is greater than or equal to 80%.
+ * final boolean healthy = diskMemoryHealthChecker.isHealthy();
+ * }</pre>
+ */
+// Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/DiskSpaceMetrics.java">DiskSpaceMetrics.java</a> in the micrometer core.
+final class DiskMemoryHealthChecker implements HealthChecker {
+
+    private final double freeDiskSpacePercentage;
+
+    private final File path;
+
+    DiskMemoryHealthChecker(double freeDiskSpacePercentage, File path) {
+        checkArgument(freeDiskSpacePercentage >= 0 && freeDiskSpacePercentage <= 1,
+                      "freeDiskSpacePercentage: %s (expected >= 0 and expected <= 1)", freeDiskSpacePercentage);
+        requireNonNull(path);
+        this.freeDiskSpacePercentage = freeDiskSpacePercentage;
+        this.path = path;
+    }
+
+    /**
+     * Returns true if the file usable space is greater or equal than the target space.
+     * @return boolean
+     */
+    @Override
+    public boolean isHealthy() {
+        return freeDiskSpacePercentage * path.getTotalSpace() <= path.getUsableSpace();
```

Comment:
`isHealthy()` could be called in an event loop and invoked frequently. On the other hand, `path.getTotalSpace()` and `path.getUsableSpace()` are blocking operations and may take some time. If there are many files in the path, the computation time will be longer.

For that reason, I propose to wrap `DiskMemoryHealthChecker` with `ScheduledHealthChecker`.

```java
final class DiskMemoryHealthChecker implements Supplier<CompletionStage<HealthCheckStatus>> {
    ...
}

static HealthChecker ofDisk(double freeDiskSpacePercentage, File path) {
    return new ScheduledHealthChecker(new DiskMemoryHealthChecker(freeDiskSpacePercentage, path), ....);
}
```

Reply:
Thanks for your comment, @ikhoon 
For now, I am available after next wednesday.
I am going to apply your comment until next weekend.
Please, refer it.

Reply:
If that's a blocking operation, we shouldn't use an event loop even if it's called periodically. What do you think of using `BlockingTaskExecutor` instead?

Reply:
Thank you for your suggestion. However, I agree with ikhoon's suggestion for now. Ideally, this PR is a health check of Disk memory space. As you said, when calling getTotalSpace() or getUsagleSpace() this is system call so it blocks other operations and takes a little some time. In this case, as we focus on disk memory state we should check periodically by scheduling. If we decide to use `BlockingTaskExecutor` it doesn't look like what I meant to verify. @minwoox What do you think?

Reply:
If an `EventLoop` is blocked, the services and clients that use the `EventLoop` are all halted.
This means that when the `EventLoop` is waiting on the `getSpace` calls, the services stopped.

While this may be acceptable when using a local file system with minimal I/O load, it can become a significant issue if the server uses network storage or experiences havey I/O load, as the blocking could last much longer.
Therefore, I believe we should use the `BlockingTaskExecutor`, which was introduced for handling blocking calls.

Reply:
I see. Thanks for your explanation. Will apply `BlockingTaskExecutor`🙂